### DISCUSSION TITLE
Signer keeps updating on every render indefinately

### DISCUSSION THREAD

**Discussion by hiteshjagwani102** [WalletConnect/walletconnect-monorepo#3975] [Developer Support]
I have been using web3ModelSigner() hook imported from @web3modal/ethers5/react. Using ethers I am creating a contract instance by passing the signer to it. I want to make sure the contract instance should update once the signer is changed so I ma using a useEffect with signer as a dependency but because signer is fetched on every render code inside useEffect runs indefinately even though the signer remains the same.

**Comment by glitch-txs** [comment]
The signer is created on every render, might be the reason, we'll remove it on the next release, there are going to be some breaking changes. We'll expose the raw provider so you can wrap it yourself with ethers.

**Reply by glitch-txs** [reply]
for now, as a workaround I'd suggest using something like the address maybe
